shell/platform/darwin/macos/framework/Source/FlutterRenderer.mm

```diff
@@ -95,30 +95,24 @@ - (FlutterMetalTexture)createTextureForView:(uint64_t)viewId size:(CGSize)size {
     // FlutterMetalTexture has texture `null`, therefore is discarded.
     return FlutterMetalTexture{};
   }
-  FlutterRenderBackingStore* backingStore = [view backingStoreForSize:size];
-  id<MTLTexture> texture = backingStore.texture;
-  FlutterMetalTexture embedderTexture;
-  embedderTexture.struct_size = sizeof(FlutterMetalTexture);
-  embedderTexture.texture = (__bridge void*)texture;
-  embedderTexture.texture_id = reinterpret_cast<int64_t>(texture);
-  return embedderTexture;
+  FlutterSurface* surface = [view.surfaceManager surfaceForSize:size];
+  FlutterMetalTexture texture = surface.asFlutterMetalTexture;
+  return texture;
```

Comment:
pedantic nit: I'd be tempted to squash the last two lines into one line, or maybe all three:
```suggestion
  return [view.surfaceManager surfaceForSize:size].asFlutterMetalTexture;
```